frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -77,36 +83,52 @@ const getColumns = () => {
   ];
 };
 
+const getNodeTaints = (node: NodeKind) => {
+  return (node.spec && node.spec.taints) || [];
+};
+
+const isNodeUnschedulable = (node: NodeKind) => {
+  const taints = getNodeTaints(node);
+  return !_.isEmpty(taints);
+};
+
+const hasOCSTaint = (node: NodeKind) => {
+  const taints = getNodeTaints(node);
+  return taints.some((taint: Taint) => _.isEqual(taint, OCS_TAINT));
+};
+
 // return an empty array when there is no data
 const getRows = (nodes: NodeKind[]) => {
-  return nodes.map((node) => {
-    const roles = getNodeRoles(node).sort();
-    const obj = {
-      cells: [],
-      selected: false,
-      id: node.metadata.name,
-      metadata: _.clone(node.metadata),
-      spec: _.clone(node.spec),
-    };
-    obj.cells = [
-      {
-        title: <ResourceLink kind="Node" name={node.metadata.name} title={node.metadata.uid} />,
-      },
-      {
-        title: roles.join(', ') || '-',
-      },
-      {
-        title: _.get(node.metadata.labels, 'failure-domain.beta.kubernetes.io/zone') || '-',
-      },
-      {
-        title: `${humanizeCpuCores(_.get(node.status, 'capacity.cpu')).string || '-'}`,
-      },
-      {
-        title: `${getConvertedUnits(_.get(node.status, 'allocatable.memory'))}`,
-      },
-    ];
-    return obj;
-  });
+  return nodes
+    .filter((node) => hasOCSTaint(node) || !isNodeUnschedulable(node))
+    .map((node) => {
+      const roles = getNodeRoles(node).sort();
+      const obj = {
+        cells: [],
+        selected: false,
+        id: node.metadata.name,
+        metadata: _.clone(node.metadata),
+        spec: _.clone(node.spec),
+      };
+      obj.cells = [
```

Comment:
I would use
```suggestion
      const cells = [
```

and then

```ts
return {
  cells,
  // ...
};
```

below. It's confusing assigning a value and reassigning it in the next line and not necessary.